File: rerun_cpp/src/rerun/components/pose_scale3d_ext.cpp
namespace rerun::components {
#if 0
    // <CODEGEN_COPY_TO_HEADER>

    /// Construct `PoseScale3D` from x/y/z values.
    PoseScale3D(float x, float y, float z) : scale{x, y, z} {}

    /// Construct `PoseScale3D` from x/y/z float pointer.
    explicit PoseScale3D(const float* xyz) : scale{xyz[0], xyz[1], xyz[2]} {}

    /// Construct a `PoseScale3D` from a uniform scale factor.
    explicit PoseScale3D(float uniform_scale) : PoseScale3D(datatypes::Vec3D{uniform_scale, uniform_scale, uniform_scale}) {}

    /// Explicitly construct a `PoseScale3D` from a uniform scale factor.
    static PoseScale3D uniform(float uniform_scale) {
        return PoseScale3D(uniform_scale);
    }

    /// Explicitly construct a `PoseScale3D` from a 3D scale factor.
    static PoseScale3D three_d(datatypes::Vec3D scale) {
        return PoseScale3D(scale);
    }

    // </CODEGEN_COPY_TO_HEADER>

#endif
} // namespace rerun::components

File: rerun_py/rerun_sdk/rerun/components/pose_scale3d_ext.py
from __future__ import annotations

from typing import TYPE_CHECKING, Any, Union

if TYPE_CHECKING:
    from rerun.datatypes import Float32Like, Vec3DLike


class PoseScale3DExt:
    """Extension for [PoseScale3D][rerun.components.PoseScale3D]."""

    def __init__(
        self: Any,
        uniform_or_per_axis: Union[Vec3DLike, Float32Like] = True,
    ):
        """
        3D scaling factor.

        A scale of 1.0 means no scaling.
        A scale of 2.0 means doubling the size.
        Each component scales along the corresponding axis.

        Parameters
        ----------
        uniform_or_per_axis:
            If a single value is given, it is applied the same to all three axis (uniform scaling).

        """
        if not hasattr(uniform_or_per_axis, "__len__") or len(uniform_or_per_axis) == 1:  # type: ignore[arg-type]
            self.__attrs_init__([uniform_or_per_axis, uniform_or_per_axis, uniform_or_per_axis])
        else:
            self.__attrs_init__(uniform_or_per_axis)
